### PR TITLE
Vale integration

### PR DESCRIPTION
**Pull request by daniel-parasoft**
This adds parsing support for results produced by vale (https://vale.sh/)
Vale is a static checker for English prose.

### Testing done

Junit test added.
We're also using this in our own infrastructure.

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    private Issue createIssue(final IssueBuilder issueBuilder, final String fileName, final JSONObject data) {
+        String checker = data.getString(CHECK);
+        String message = data.getString(MESSAGE_KEY);
+        String severity = data.getString(SEVERITY_KEY);
+        String link = data.getString(LINK_KEY);
+        int line = data.getInt(LINE_KEY);
+        JSONArray span = data.getJSONArray(SPAN_KEY);
+        int startColumn = span.getInt(0);
+        int endColumn = span.getInt(1);
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java:51]: Can you inline these variables into the builder method parameter, there is no need for a variable.

File: src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                analysisSeverity = Severity.ERROR;
+                break;
+            case "warning":
+                analysisSeverity = Severity.WARNING_NORMAL;
+                break;
+            case "suggestion":
+                analysisSeverity = Severity.WARNING_LOW;
+                break;
+            default:
+                analysisSeverity = Severity.WARNING_NORMAL;
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java:64]: Extract this part into a method that returns the severity. 

Or check if it makes sense to add `"suggestion"` to `Severity.guessSeverity` and reuse the existing code.

File: SUPPORTED-FORMATS.md [hunk 3]
```diff
@@ -2178,6 +2178,27 @@ analyze - iccxxxxcompiler_opts cstat2.c</pre></code>For details check the IAR C-
...
+            </td>
+            <td>
+                **/vale-report.json
+            </td>
+        </tr>
+        <tr>
+            <td colspan="4">
+                :bulb: Use option --output=JSON
+            </td>
+        </tr>
```
Comment (uhafner) [inline SUPPORTED-FORMATS.md:2201]: Revert this part as this file is generated (on a merge to main).

**Review by uhafner — CHANGES_REQUESTED** [review]
Thanks for your PR! 

Just some minor things...

File: src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java (hunk 1 above)
Comment (daniel-parasoft) [reply src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java]: Done

File: src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java (hunk 2 above)
Comment (daniel-parasoft) [reply src/main/java/edu/hm/hafner/analysis/parser/ValeParser.java]: Done

File: SUPPORTED-FORMATS.md (hunk 3 above)
Comment (daniel-parasoft) [reply SUPPORTED-FORMATS.md]: Done

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
Thanks for your PR!

**Pull request merged**